.github/actions/pr-linting/index.js

```diff
@@ -0,0 +1,29 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+run();
+async function run() {
+  const title = core.getInput("pr-title")
+
+
+  const context = github.context;
+  if (context.payload.pull_request == null) {
+      core.setFailed('No pull request found.');
+      return;
+  }
+
+  core.info(
+    `🔎 Checking if the title of this PR "${title}" meets the requirements ...`
+  );
+
+  if ((/^\[?wip\]?/i).test(title)) {
+    core.info('This PR is marked as a WIP. Skipping validation');
+  } 
+  else if ((/^(fix|feat|build|chore|docs|test|refactor|ci|localize|bump|revert)(\(([A-Za-z])\w+\))?:/gmi).test(title)) {
```

Comment:
I'm a little concerned that some of these feel a bit similar (ci vs build) and/or perhaps too granular (test/refactor/bump/localize feel like they primary are updates done in tandem with larger fixes/features and may not be used too often). Happy to be overruled since I'm not familiar with how other repos handle these classifications, but it feels like maybe some of these could be placed under a generic prefix as part of its scope (e.g. fix(test) or chore(bump), etc) Maybe scope could be generalized and instead contain any supplementary information instead (e.g. fix(RAC, localize) Update ColorPicker intl strings to fix crash)

Reply:
happy to start with a large set or a small set and see how it goes, can always update down the line